src/faucet/server_api.rs

```diff
@@ -189,3 +189,137 @@ pub async fn signed_erc20_transfer(
     let signed = sign_with_eth_secret_key(tx.clone(), faucet_info).await?;
     Ok(signed)
 }
+
+#[server(endpoint = "claim_token", input = GetUrl)]
+pub async fn claim_token(
+    faucet_info: FaucetInfo,
+    address: String,
+) -> Result<String, ServerFnError> {
+    use crate::utils::address::{AddressAlloyExt, parse_address};
+    use crate::utils::message::message_transfer;
+    use crate::utils::rpc_context::Provider;
+    use fvm_shared::address::Network;
+    use fvm_shared::address::set_current_network;
+    use leptos_axum::ResponseOptions;
+    use send_wrapper::SendWrapper;
+
+    let response = leptos::prelude::expect_context::<ResponseOptions>();
+
+    SendWrapper::new(async move {
+        match faucet_info {
+            FaucetInfo::MainnetFIL => {
+                response.set_status(http::StatusCode::IM_A_TEAPOT);
+                Err(ServerFnError::ServerError(
+                    "I'm a teapot - mainnet tokens are not available.".to_string(),
+                ))
+            }
+            FaucetInfo::CalibnetFIL => {
```

Comment:
Instead of handling each case here, you can create a separate function: `handle_native_claim` and `handle_erc20_claim`, it will become little bit easier to read.